src/sql/SqlStatement.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Long from 'long';
+
+export enum SqlExpectedResultType {
+    ANY,
+    ROWS,
+    UPDATE_COUNT
+}
+
+export type SqlExpectedResultTypeStrings = keyof typeof SqlExpectedResultType;
+
+/**
+ * Options used to change behavior of an SQL query.
+ */
+export interface SqlStatementOptions {
+    /**
+     * The schema name. The engine will try to resolve the non-qualified object identifiers from the statement in the
+     * given schema. If not found, the default search path will be used, which looks for objects in the predefined
+     * schemas `partitioned` and `public`.
+     *
+     * * The schema name is case sensitive. For example, `foo` and `Foo` are different schemas.
+     * * The default value is `null` meaning only the default search path is used.
+     */
+    schema?: string;
```

Comment:
should this have the type of `string | null`?

Reply:
BTW, let me refresh my TypeScript knowledge :smile: `schema?: string` means it can either be undefined or string right? I think we can be fine with that

Reply:
you are right. just forgot to change it, it should be schema?: string | null;

Reply:
no I have already changed it :D

Reply:
Sorry for not understanding this earlier. I have rusty TypeScript knowledge. 

Maybe it is more work for you but I think we should aim for the best possible/good-looking API for the users. I think if you also feel that `schema?: string` is better/more idiomatic than `schema?: string | null`, I vote for reverting these changes but I leave the decision to you

Reply:
The behavior change is allowing null or not. You proposed allowing null. ? means this option is optional. If it is not provided we will not validate it and use default schema. Do you propose disallowing null?

Reply:
Yes. I didn't understand this was optional initially. So, instead of `null`, they will not provide it at all as you suggested. This is more common in JS right rather than setting something to `null`